processing/src/main/java/io/druid/segment/filter/ExtractionFilter.java

```diff
@@ -57,12 +55,13 @@ public ExtractionFilter(
     if (allDimVals != null) {
       for (int i = 0; i < allDimVals.size(); i++) {
         String dimVal = allDimVals.get(i);
-        if (value.equals(fn.apply(dimVal))) {
+        if (value.equals(Strings.nullToEmpty(fn.apply(dimVal)))) {
```

Comment:
value can not be null thanks to [this line of code](https://github.com/druid-io/druid/blob/61a5bb27847f329fe28b32c791bf3a4a9d315b45/processing/src/main/java/io/druid/query/filter/ExtractionDimFilter.java#L46)